src/EditorFeatures/Core.Wpf/InlineRename/UI/Dashboard/RenameDashboard.xaml.cs

```diff
@@ -243,7 +248,9 @@ protected override void OnAccessKey(AccessKeyEventArgs e)
                 }
                 else if (string.Equals(e.Key, RenameShortcutKey.Apply, StringComparison.OrdinalIgnoreCase))
                 {
-                    this.Commit();
+                    var token = _listener.BeginAsyncOperation($"{nameof(OnAccessKey)}.{nameof(RenameShortcutKey.Apply)}");
+                    // CommitAsync catch & report all the exceptions.
+                    _ = this.CommitAsync().CompletesAsyncOperation(token);
```

Comment:
if htis is so, then why does RenameFlyout to ReportNonFatalErrorAsync()?

my recommendation is to have one synchronous method (like InitiateCommit) that everyone calls into that has consistent behavior :)